### PR TITLE
Show lobby timer and description in lobby screen 

### LINKED ISSUE #1557: Show lobby timer and description in lobby screen
Server ref https://github.com/nextcloud/spreed/issues/4921

Information should be available already via the room api, so just needs frontend work?

### PR DESCRIPTION
**Pull request by timkrueger**
resolve #1557

I didn't get it the way like in [iOS](https://github.com/nextcloud/talk-ios/pull/615#issue-987829178). So stay with the timer like in spreed. Is that OK?

Days | Hours | Minutes | Soon
------- | -------- | ---------- | --------
days | hours | minutes | soon

### CODE REVIEW

**Comment by mahibi** [comment]
i'm a bit confused by the german text. something is wrong here with translations or concatenation..?

grafik

**Comment by timkrueger** [comment]
> i'm a bit confused by the german text. something is wrong here with translations or concatenation..?
> 
> grafik

I'll take a look

File: app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt [hunk 1]
```diff
@@ -1119,19 +1119,20 @@ class ChatController(args: Bundle) :
...
                         resources!!.getString(R.string.nc_lobby_waiting_with_date),
-                        DateUtils.getLocalDateStringFromTimestampForLobby(
-                            currentConversation?.lobbyTimer
-                                ?: 0
-                        )
+                        DateUtils.relativeStartTimeForLobby(timestamp, resources!!)
                     )
                 } else {
-                    binding.lobby.lobbyTextView.setText(R.string.nc_lobby_waiting)
+                    resources!!.getText(R.string.nc_lobby_waiting)
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt:1131]: resources?.getText

File: app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt [hunk 2]
```diff
@@ -41,6 +54,48 @@ object DateUtils {
...
+                }
+                minutes > 0 -> {
+                    fmt.format(
+                        minutes.toDouble(),
+                        Direction.NEXT,
+                        RelativeUnit.MINUTES
+                    )
+                }
+                else -> {
+                    resources.getString(R.string.nc_lobby_start_soon)
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt:92]: the soon case should not be done the same way the relative date works. As Marcel pointed out it breaks with translations

Comment (timkrueger) [reply app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt:92]: > the soon case should not be done the same way the relative date works. As Marcel pointed out it breaks with translations

You're right, of course. I'll change it.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -350,7 +350,9 @@
     <string name="nc_lobby">Lobby</string>
     <string name="nc_start_time">Start time</string>
     <string name="nc_lobby_waiting">You are currently waiting in the lobby.</string>
-    <string name="nc_lobby_waiting_with_date">You are currently waiting in the lobby.\n This meeting is scheduled for %1$s.</string>
+    <string name="nc_lobby_waiting_with_date">You are currently waiting in the lobby.\n\nThis meeting is scheduled for
```
Comment (mahibi) [inline app/src/main/res/values/strings.xml:353]: i think we should avoid linebreaks in translations. better would be to have 2 seperate strings?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -350,7 +350,9 @@
...
+        %1$s.</string>
+    <string name="nc_lobby_start_soon">The meeting will start soon</string>
```
Comment (mahibi) [inline app/src/main/res/values/strings.xml:355]: not used anymore?

File: app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt (hunk 2 above)
Comment (timkrueger) [reply app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt:97]: I left the soon case in the else part of the relative dates handling but append the complete text in the lobby screen in an other way.

**Review by mahibi — APPROVED** [review]

**Pull request merged**